### ISSUE TITLE
Double forward slash on requests 

### ISSUE THREAD

**Issue by Cmarenburg** [meilisearch/meilisearch-php#29]
Just curious if anyone is experiencing routes being generate with a double forward slash on requests (ending up within incoming requests to the Meilisearch server like POST //indexes//documents?primaryKey=id (taken from the console), it's resulting in 404's not found with the server, when built with the binary.
I'm not really sure where to go from here. When I generate a HTTP request via a rest client it all works fine. Anyone else have any issues?

Can’t remember the version I’m running. Don’t have it at hand.

Sorry didn’t realize it was related to the core.

**Comment by curquiza** [comment]
@Kerollmops this issue is still on `master` 😕
For the slash at this end and in the middle of the URL.

**Comment by curquiza** [comment]
@Kerollmops for me it's not a PHP issue related since it does not work with `curl`

**Comment by Kerollmops** [comment]
Yeah, malformed request urls doens't work with curl nether, I think there is three things here:
  - This is a normal behavior to refuse urls with double slashes.
  - It is invalid behavior to refuse urls ending with a slash.
  - If this is the PHP client which generate those urls, we must fix it.

I invoque @qdequele for the latter 🐉

**Comment by curquiza** [comment]
About the 3rd point: all the methods in the SDK packages are tested, so there is no unexpected slash generation. The only way for the user to add an unexpected slash is during the client declaration:

```php
$client = new MeiliSearch\Client('http://127.0.0.1:7700/');
// instead of
$client = new MeiliSearch\Client('http://127.0.0.1:7700');
```

It will not work because MeiliSearch API does not handle the double slashes.

If MeiliSearch will not plan to handle that (GitHub handles it for example), I will open an issue to remove the unexpected slash during client declaration in the SDKs, because they all have the same problem 😉

**Comment by qdequele** [comment]
We have the solution here: https://docs.rs/actix-web/3.0.0-alpha.1/actix_web/middleware/struct.NormalizePath.html 😉

**Comment by Kerollmops** [comment]
Does it also remove the trailling slash?

**Comment by qdequele** [comment]
No, I will make a PR on this structure. Actix seems to be very open to PR.

**Comment by curquiza** [comment]
@Kerollmops can you move back this issue to MeiliSearch repo then? 🙂
